packages/jellyfish-transaction/src/script/dftx/dftx_loans.ts

```diff
@@ -36,6 +36,17 @@ export interface SetCollateralToken {
   activateAfterBlock: number // ---| 4 bytes unsigned, Changes will be active after the block height
 }
 
+/**
+ * SetLoanToken DeFi Transaction
+ */
+export interface SetLoanToken {
+  symbol: string // ------------| c = VarUInt{1-9 bytes}, + c bytes UTF encoded string, Symbol or id of collateral token
+  name: string // --------------| c = VarUInt{1-9 bytes}, + c bytes UTF encoded string, Token's name, no longer than 128 characters
+  priceFeedId: string // -------| 32 bytes, hex string, Txid of oracle feeding the price
+  mintable: boolean // ---------| 4 bytes, mintable, Token's 'Mintable' property
+  interest: BigNumber // -------| 8 bytes unsigned, interest rate
```

Comment:
```suggestion
  interest: BigNumber // -------| 8 bytes signed, interest rate
```
understand negative interest is not allowed but the `ain` is a signed int

Reply:
In the C++ code, the variable for interest is CAmount. I think this should be unsigned right?

I did a testing just now on this variable and I passed in a negative value. This is what I got

The value of "value" is out of range. It must be >= 0 and <= 4294967295. Received -1512345678RangeError [ERR_OUT_OF_RANGE]: The value of "value" is out of range. It must be >= 0 and <= 4294967295. Received -1512345678

Reply:
`CAmount` is int64_t.. it can be negative

Reply:
@jingyi2811 the error you are getting should come from some validation. But CAmount is `int64_t`, we should change accordingly.

Reply:
So far, some / all the documentation for ComposableBuffer.satoshiAsBigNumber are unsigned. I guess we may park this fix in another PR.

Reply:
ok some history on this one.
@Jouzo asked me about CAmount being negative sometimes back. My answer was that it's merely for the sake of arithmetic operations. I am not 100% sure if we ever use negative CAmount in an actual use case. But inside `defid` there are explicit checks to make sure CAmout is positive.
https://github.com/DeFiCh/ain/blob/89de3bb4adddcb2dd028f41c34009a1d9958c127/src/amount.h#L118
Also, these checks would be meaningless if CAmount is unsigned.

Reply:
From what I gathered on `ain`, CAmount is always positive when used in combination with MoneyRange:
-  See https://github.com/DeFiCh/ain/blob/89de3bb4adddcb2dd028f41c34009a1d9958c127/src/amount.h#L191
- https://github.com/DeFiCh/ain/blob/89de3bb4adddcb2dd028f41c34009a1d9958c127/src/test/amount_tests.cpp#L15
or after some arithmetic operations :
- See https://github.com/DeFiCh/ain/blob/89de3bb4adddcb2dd028f41c34009a1d9958c127/src/amount.h#L132
- or https://github.com/DeFiCh/ain/blob/89de3bb4adddcb2dd028f41c34009a1d9958c127/src/amount.h#L119
This is tested in amount_test.cpp:
- See https://github.com/DeFiCh/ain/blob/89de3bb4adddcb2dd028f41c34009a1d9958c127/src/test/amount_tests.cpp#L128

In amount_test.cpp, there is reference to CAmount as negative value but it seems to be used to handle feeRate in an odd way :
- See https://github.com/DeFiCh/ain/blob/89de3bb4adddcb2dd028f41c34009a1d9958c127/src/test/amount_tests.cpp#L41

In our case, I think it is safe to assume that CAmount will always be positive and we can keep on using `ComposableBuffer.satoshiAsBigNumber` for what is CAmount on `ain`